src/common/timing/time-keeper.js

```diff
@@ -30,18 +35,25 @@ export class TimeKeeper {
    */
   #ready = false
 
-  constructor () {
-    this.#originTime = Date.now() - performance.now()
+  constructor (agentIdentifier) {
+    this.#session = getRuntime(agentIdentifier)?.session
+
+    if (this.#session) {
+      const ee = baseEE.get(agentIdentifier)
+      ee.on(SESSION_EVENTS.UPDATE, this.#processSessionUpdate.bind(this))
+      ee.on(SESSION_EVENTS.STARTED, () => {
```

Comment:
Does this only matter for "reset" sessions?  It will inherently miss the first started event, but if that is true would we still be safer to check the arg `isNew`?

Reply:
maybe it doesnt matter since it already checks itself for #ready as well

Reply:
It is better that it checks `#ready` and is only reliant on itself. It doesn't need to capture the first call on an absolutely brand new session since the next statement just reads the state. This is meant just to capture session timeouts. I was going to use the `RESET` event but the session manager completely replaces the session state after all the reset handlers are ran.